examples/classic_controllers/classic_controllers.py

```diff
@@ -1471,6 +1598,595 @@ def reset(self):
+
+        labels = [{'y_label': r"|$\Psi_{r}$|/Vs", 'state_label': r"|$\hat{\Psi}_{r}$|/Vs"},
+                  {'y_label': r"$\measuredangle\Psi_r$/rad", 'state_label': r"$\measuredangle\hat{\Psi}_r$/rad"}]
+
+        for ext_plot, label in zip(self.external_plot, labels):
+            ext_plot.set_label(label)
+
+        if self.has_cont_action_space:
+            assert len(stages[0]) == 2, 'Number of stages not correct'
+            self.decoupling = controller_kwargs.get('decoupling', True)
+            self.u_sd_0 = self.u_sq_0 = 0
+            self.d_controller = _controllers[stages[0][0]['controller_type']][1].make(
+                environment, stages[0][0], **controller_kwargs)
+            self.q_controller = _controllers[stages[0][1]['controller_type']][1].make(
+                environment, stages[0][1], **controller_kwargs)
+
+    def control(self, state, reference):
+        """
+            This is the main methode of the FieldOrientedControllerRotorFluxObserver. It calculates the input voltages
+            u_a,b,c
+        """
+        state = state * self.limits
+        psi_abs, psi_angle = self.flux_observer.estimate(state)
+        omega_me = state[self.omega_idx]
+        i_sd = state[self.i_sd_idx]
+        i_sq = state[self.i_sq_idx]
+        omega_s = omega_me + self.r_r * self.l_m / self.l_r * i_sq / max(np.abs(psi_abs), 1e-4) * np.sign(psi_abs)
+
+        if self.decoupling:
+            self.u_sd_0 = -omega_s * self.sigma * self.l_s * i_sq - self.l_m * self.r_r / (self.l_r ** 2) * psi_abs
+            self.u_sq_0 = omega_s * self.sigma * self.l_s * i_sd + omega_me * self.l_m / self.l_r * psi_abs
+
+        u_sd_delta = self.d_controller.control(state[self.i_sd_idx], reference[self.i_sd_ref_idx] * self.limits[self.i_sd_idx])
+        u_sq_delta = self.q_controller.control(state[self.i_sq_idx], reference[self.i_sq_ref_idx] * self.limits[self.i_sq_idx])
+
+        u_sd = self.u_sd_0 + u_sd_delta
+        u_sq = self.u_sq_0 + u_sq_delta
+
+        u_s_abc = self.dq_to_abc_transformation((u_sd, u_sq), psi_angle)
+
+        u_s_abc /= self.limits[self.u_s_abc_idx]
+
+        action = np.clip(u_s_abc, self.action_space.low, self.action_space.high)
+
+        if action.all() == u_s_abc.all():
+            self.d_controller.integrate(state[self.i_sd_idx], reference[self.i_sd_ref_idx] * self.limits[self.i_sd_idx])
+            self.q_controller.integrate(state[self.i_sq_idx], reference[self.i_sq_ref_idx] * self.limits[self.i_sq_idx])
+        self.psi_abs = psi_abs
+        self.psi_angle = psi_angle
+        self.plot(external_plot=self.external_plot)
+        return action
+
+    def get_plot_data(self):
+        return dict(ref_state=[], ref_value=[], external=[[self.psi_abs], [self.psi_angle]])
+
+    def reset(self):
+        self.flux_observer.reset()
+        self.d_controller.reset()
+        self.q_controller.reset()
+
+
+class CascadedFieldOrientedControllerRotorFluxObserver(Controller):
```

Comment:
I think, the naming of all new classes qith the postfix "RotorFluxObserver" is a little imprecise. This is a controller and not an observer.

Maybe it could be named `InductionMotorFieldOrientedController`  or even `RotorFluxObserverFieldOrientedController` @max-schenke what do you think about it?